### PR TITLE
Avoid if in the inner loop

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst PR

### CODE REVIEW

**Review by geirst — APPROVED** [review]
👍 Consider adding more test cases

File: searchlib/src/tests/features/prod_features.cpp [hunk 1]
```diff
@@ -1225,8 +1227,8 @@ Test::testDotProduct()
         assertDotProduct(17, "(0:1,3:4,50:97)", 1, "sint", "arrfloat"); // attribute override
         assertDotProduct(0, "(0:1,3:4,50:97)", 1, "sint", "arrfloat_non_existing"); // incorrect attribute override
     }
-    verifyCorrectDotProductExecutor(_factory, "wsstr", "search::features::dotproduct::wset::(anonymous namespace)::DotProductExecutorByEnum");
-    verifyCorrectDotProductExecutor(_factory, "wsint", "search::features::dotproduct::wset::DotProductExecutor<search::MultiValueNumericAttribute<search::IntegerAttributeTemplate<int>, search::multivalue::WeightedValue<int> > >");
+    verifyCorrectDotProductExecutor(_factory, "wsstr", "{a:1}", "search::features::dotproduct::wset::(anonymous namespace)::DotProductExecutorByEnum");
+    verifyCorrectDotProductExecutor(_factory, "wsint", "{1:1}", "search::features::dotproduct::wset::DotProductExecutor<search::MultiValueNumericAttribute<search::IntegerAttributeTemplate<int>, search::multivalue::WeightedValue<int> > >");
```
Comment (geirst) [inline searchlib/src/tests/features/prod_features.cpp]: Consider adding test cases where the SingleZeroValueExecutor is returned as well.

Comment (baldersheim) [reply searchlib/src/tests/features/prod_features.cpp]: Done

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**